diesel/src/types/ord.rs

```diff
@@ -14,3 +16,5 @@ impl SqlOrd for types::Interval {}
 impl SqlOrd for types::Time {}
 impl SqlOrd for types::Timestamp {}
 impl<T: SqlOrd + NotNull> SqlOrd for types::Nullable<T> {}
+#[cfg(feature = "postgres")]
+impl<T: SqlOrd> SqlOrd for pg::types::sql_types::Array<T> {}
```

Comment:
You don't need to go through the `pg` module for this. Just `types::Array<T>` is enough.